components/org.wso2.carbon.identity.oauth/src/main/java/org/wso2/carbon/identity/oauth2/validators/OAuth2JWTTokenValidator.java

```diff
@@ -177,11 +178,32 @@ private IdentityProvider getResidentIDPForIssuer(String jwtIssuer) throws Identi
     }
 
     private boolean validateSignature(SignedJWT signedJWT, IdentityProvider idp)
-            throws JOSEException, IdentityOAuth2Exception {
+            throws JOSEException, IdentityOAuth2Exception, ParseException {
 
         JWSVerifier verifier = null;
+        X509Certificate x509Certificate = null;
         JWSHeader header = signedJWT.getHeader();
-        X509Certificate x509Certificate = resolveSignerCertificate(header, idp);
+        JWTClaimsSet jwtClaimsSet = signedJWT.getJWTClaimsSet();
+
+        Map<String, String> realm = (HashMap) jwtClaimsSet.getClaim(OAuthConstants.OIDCClaims.REALM);
+
+        // Get certificate from tenant if available in claims.
+        if (realm != null && !realm.isEmpty()) {
```

Comment:
You can use CollectionsUtil.isNotEmpty() 
```suggestion
        if (CollectionUtils.isNotEmpty(realm)) {
```

Reply:
Cannot use CollectionUtils, used MapUtils instead. Thanks for the approach.